lighthouse-core/audits/resource-summary.js

```diff
@@ -0,0 +1,156 @@
+ */
+'use strict';
+
+const Audit = require('./audit');
+const NetworkRecords = require('../computed/network-records.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainResource = require('../computed/main-resource.js');
+const URL = require('../lib/url-shim');
+const Util = require('../report/html/renderer/util.js');
+
+const UIStrings = {
+  /** Imperative title of a Lighthouse audit that tells the user to minimize the size and quantity of resources used to load the page. */
+  title: 'Keep request counts and file sizes small',
+  /** Description of a Lighthouse audit that tells the user that they can setup a budgets for the quantity and size of page resources. No character length limits. */
+  description: 'To set budgets for the quantity and size of page resources,' +
+    ' add a budgets.json file.',
+  /** [ICU Syntax] Label for an audit identifying the number of requests and kilobytes used to load the page. */
+  displayValue: `{requestCount, plural, =1 {1 request} other {# requests}} • { formattedByteCount}`,
+  /** Label for the combination of all resources on the page. */
+  totalResourceType: 'Total',
+  /** Label for the 'Document' resource type. */
+  documentResourceType: 'Document',
+  /** Label for the 'script' resource type. */
+  scriptResourceType: 'Script',
+  /** Label for the 'stylesheet' resource type. */
+  stylesheetResourceType: 'Stylesheet',
+  /** Label for the 'image' resource type. */
+  imageResourceType: 'Image',
+  /** Label for the 'media' resource type. */
+  mediaResourceType: 'Media',
+  /** Label for the 'font' resource type. */
+  fontResourceTyoe: 'Font',
+  /** Label for the 'other' resource type. */
+  otherResourceType: 'Other',
+  /** Label for the 'third-party' resource type. */
+  thirdPartyResourceType: 'Third-party',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class ResourceSummary extends Audit {
+  /**
+       * @return {LH.Audit.Meta}
+       */
+  static get meta() {
+    return {
+      id: 'resource-summary',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      requiredArtifacts: ['devtoolsLogs'],
+    };
+  }
+
+  /**
+     * @param {LH.Artifacts.NetworkRequest} record
+     * @return string
+     */
+  static determineResourceType(record) {
+    switch (record.resourceType) {
+      case 'Stylesheet':
+      case 'Image':
+      case 'Media':
+      case 'Font':
+      case 'Script':
+      case 'Document':
+        // budgets.json uses lowercase for resource types, unlike NetworkRequest.resourceType
+        return record.resourceType.toLowerCase();
+      default:
+        return 'other';
+    }
+  }
+
+  /**
+     * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+     * @param {string} mainResourceURL
+     * @return {Object<string,{label: string, resourceType: string, count: number, size: number}>}
```

Comment:
@brendankenny knows better if these translate too (IIRC they might just become `any` like how `Object` works in TS?), but the TS way we've mostly adopted elsewhere `Record<TKey, TValue>`